sia/upload_test.go

```diff
@@ -0,0 +1,42 @@
+package sia
+
+import (
+	"testing"
+	"time"
+
+	"github.com/NebulousLabs/Sia/sia/components"
+)
+
+func testUploadFile(t *testing.T, c *Core) {
+	if testing.Short() {
+		return
+	}
+	// Check that hostDB has at least one entry.
+	if c.hostDB.Size() < 1 {
+		t.Fatal("Hostdb needs at least 1 host to perform testUploadFile")
+	}
+
+	// Have the renter negotiate a contract with the host in the hostDB.
+	err := c.renter.RentFile(components.RentFileParameters{
+		Filepath:    "sia.go",
+		Nickname:    "one",
+		TotalPieces: 1,
+	})
+	if err != nil {
+		t.Error(err)
+	}
+
+	time.Sleep(10 * time.Second)
```

Comment:
I'm not sure why it takes 10 seconds but 5 wasn't working 100% of the time.

This is the wrong approach anyway and a hack. 10 seconds means there's decent motivation to actually fix it.